cmake/Dependencies.cmake

```diff
@@ -968,7 +971,7 @@ if(USE_GLOO)
     # Add explicit dependency since NCCL is built from third_party.
     # Without dependency, make -jN with N>1 can fail if the NCCL build
     # hasn't finished when CUDA targets are linked.
-    if(USE_NCCL)
+    if(USE_NCCL AND NOT USE_SYSTEM_NCCL)
```

Comment:
Should `gloo_cuda` (the line below) be handled differently?

Reply:
For ROCm builds, we do not support the "nccl_external" (third_party/nccl) option currently, so we don't need to have that additional dependency for "gloo_hip". "gloo_cuda" is only relevant for CUDA builds anyway. Does that clarify?

Reply:
@xuhdev Should I change the condition to be `if(USE_NCCL AND NOT USE_ROCM)`?

Reply:
If `gloo_cuda` doesn't apply, I think you wanna make sure `USE_ROCM` is false.